packages/UI/src/components/Switch/Switch.tsx

```diff
@@ -0,0 +1,62 @@
+import * as SwitchPrimitives from '@radix-ui/react-switch';
+import * as React from 'react';
+import { cn } from '../../utils/classesMerger';
+
+interface SwitchProps
+  extends Omit<
+    React.ComponentPropsWithoutRef<typeof SwitchPrimitives.Root>,
+    'asChild'
+  > {
+  /**
+   * The controlled state of the switch. Must be used in conjunction with onCheckedChange.
+   */
+  checked?: SwitchPrimitives.SwitchProps['checked'];
+  /**
+   * The state of the switch when it is initially rendered. Use when you do not need to control its state.
+   */
+  defaultChecked?: SwitchPrimitives.SwitchProps['defaultChecked'];
+  /**
+   * When true, indicates that the user must check the switch before the owning form can be submitted.
+   */
+  required?: SwitchPrimitives.SwitchProps['required'];
+  /**
+   * Event handler called when the state of the switch changes.
+   */
+  onCheckedChange?: SwitchPrimitives.SwitchProps['onCheckedChange'];
+  /**
+   * When true, prevents the user from interacting with the switch.
+   */
+  disabled?: SwitchPrimitives.SwitchProps['disabled'];
+  /**
+   * The name of the switch. Submitted with its owning form as part of a name/value pair.
+   */
+  name?: SwitchPrimitives.SwitchProps['name'];
+  /**
+   * The value given as data when submitted with a name.
+   */
+  value?: SwitchPrimitives.SwitchProps['value'];
+}
+
+/**
+ * A control that allows the user to toggle between checked and not checked.
+ * @see http://localhost:6006/?path=/docs/components-switch--docs
+ * @see https://www.radix-ui.com/primitives/docs/components/switch#root
+ */
+const Switch = React.forwardRef<
+  React.ElementRef<typeof SwitchPrimitives.Root>,
+  SwitchProps
+>(({ className, ...props }, ref) => (
+  <SwitchPrimitives.Root
+    className={cn(
+      'peer inline-flex h-6 w-12 shrink-0 cursor-pointer items-center rounded-2xl outline outline-1 -outline-offset-1 outline-grayscale-800 transition-colors disabled:cursor-not-allowed disabled:!bg-grayscale-100 disabled:!outline-GreyScale-charcoal data-checked:bg-grayscale-800 data-unchecked:bg-grayscale-50',
```

Comment:
Because of the outline-1 don't you need a m-1?

Also what does the ! do in "disabled:!bg-grayscale-100" ?

Reply:
why do you think I need m-1?

! in "disabled:!bg-grayscale-100" adds important to that rule.

https://tailwindcss.com/docs/configuration#important-modifier

Reply:
I was mistaken, it was with "outline-2" that you need "m-1."
Outline is outside the element, it will look misaligned at a certain point and needs margins in order to align correctly.

Reply:
One other question is why are we using outline here instead of border?

Reply:
> One other question is why are we using outline here instead of border?

because I need to use it as "inner border" that's why I have accompanied it with  `-outline-offset-1` which basically puts the outline 1px inside the content box

Reply:
outline offset is something you can't do with border but you have to know that outline is also not calculated as part the box-model size where border is calculated.